### ISSUE TITLE
Console tab shows no text at engine start with Cached UI enabled

### ISSUE THREAD

**Issue by raphael-ourmachinery** [OurMachinery/themachinery-public#652]
OS: ArchLinux
Window Manager: I3wm

Steps to reproduce:
- Start the engine;
- No text will appears in console tab, only appears after to hover mouse over console tab

![Screenshot from 2021-07-23 10-21-24](https://user-images.githubusercontent.com/75702792/126788429-a44dba68-ad3b-4f0b-855b-28d62da3b01f.png)

**Comment by raphael-ourmachinery** [comment]
Just a update, I couldn't reproduce with these steps, but doing the following I have a similar issue:
- Open Gameplay Third Person sample;
- Start a simulation;
- Drag the simulation to a new window;
- Save and close;
- When open the engine again, text isn't showing correct on console tab
![Screenshot from 2021-07-28 09-58-40](https://user-images.githubusercontent.com/75702792/127326065-2c2cd74a-2037-4902-be35-6bbd4975fc4f.png)

**Comment by niklas-ourmachinery** [comment]
Your repro case doesn't work for me @raphael-ourmachinery.

I think this is not an OS specific bug, because I've seen it happen on Windows too. But I don't have any specific repro case, it seems to happen randomly on startup.

It can sometimes happen with other tabs too, I've seen it in the Asset Browser, though the console tab tends to be the most common one.

I think the root of the problem is that we generate the font data every frame in `update_editor_gui()`. Some of that data is stored in the `vbuffer` of the UI.

When we are drawing a tab, it refers to this stored data. When caching is enabled, these references end up in the cache.

So, if the font data at the start of the `vbuffer` changes for some reason, the references to it in the cached data will no longer be valid and we end up with text that is garbled in various ways.

A potential fix for this would be to compute a hash of the initial font data that goes into the `vbuffer` -- and if that hash changes from frame to frame, then we invalidate all the tab caches.

Buuut... we should also maybe look into it so that we don't regenerate the font data every frame... not sure what you think about that @tobias-ourmachinery ?

**Comment by niklas-ourmachinery** [comment]
Logging the hash of the vbuffer, I can see that there is a change in the first frame and after that it is stable, so looks like this is a likely explanation:

~~~
vbuffer_hash: da697d3b1f9b04f9
vbuffer_hash: 2dd84c6cf05fb5a (205)
~~~

**Comment by niklas-ourmachinery** [comment]
I've pushed a potential fix for this that will go out in 2021.7.

Without a reliable repro case, I can't be 100 % sure that this fixes the issue, but I will close this as fixed for now. If it pops up again, reopen or create a new issue.

**Comment by raphael-ourmachinery** [comment]
I tested it and I'm not able to reproduce it anymore.
